src/sec_certs/dataset/cc.py

```diff
@@ -273,12 +272,34 @@ def _set_local_paths(self):
                 self.certificates_txt_dir,
             )
 
-    def process_auxiliary_datasets(self, download_fresh: bool = False) -> None:
-        self.aux_handlers[CCMaintenanceUpdateDatasetHandler].certs_with_updates = [  # type: ignore
-            x for x in self if x.maintenance_updates
-        ]
-        self.aux_handlers[CCSchemeDatasetHandler].only_schemes = {x.scheme for x in self}  # type: ignore
-        super().process_auxiliary_datasets(download_fresh)
+    def process_auxiliary_datasets(
+        self,
+        download_fresh: bool = False,
+        processed_pp_dataset_root_dir: Path | None = None,
+        skip_schemes: bool = False,
+        **kwargs,
+    ) -> None:
+        if CCMaintenanceUpdateDatasetHandler in self.aux_handlers:
+            self.aux_handlers[CCMaintenanceUpdateDatasetHandler].certs_with_updates = [  # type: ignore
+                x for x in self if x.maintenance_updates
+            ]
+        if CCSchemeDatasetHandler in self.aux_handlers:
+            self.aux_handlers[CCSchemeDatasetHandler].only_schemes = {x.scheme for x in self}  # type: ignore
+
+        if processed_pp_dataset_root_dir:
+            if self.aux_handlers[ProtectionProfileDatasetHandler].root_dir.exists():
+                logger.warning(
+                    f"Overwriting PP Dataset at {self.aux_handlers[ProtectionProfileDatasetHandler].root_dir} with dataset from {processed_pp_dataset_root_dir}."
+                )
+            shutil.copytree(
+                processed_pp_dataset_root_dir,
+                self.aux_handlers[ProtectionProfileDatasetHandler].root_dir,
+                dirs_exist_ok=True,
+            )
+
+        if skip_schemes:
+            del self.aux_handlers[CCSchemeDatasetHandler]
```

Comment:
Is it a good idea for it to delete the handler? It modifies the dataset state and I think that is quite unexpected.

Reply:
Thanks for the tip. I needed something to commit on yesterday, I want to revisit this. It's a bad idea and I'm thinking about better design atm.